src/Tools/Source/RunTests/AssemblyScheduler.cs

```diff
@@ -2,359 +2,314 @@
+
+        private static void LogWorkItems(ImmutableArray<WorkItemInfo> workItems)
         {
-            if (type.BaseType.Kind != HandleKind.TypeReference)
+            ConsoleUtil.WriteLine($"Built {workItems.Length} work items");
+            Logger.Log("==== Work Item List ====");
+            foreach (var workItem in workItems)
             {
-                return false;
+                var totalExecutionTime = TimeSpan.FromMilliseconds(workItem.Filters.Values.SelectMany(f => f).Sum(f => f.ExecutionTime.TotalMilliseconds));
+                Logger.Log($"- Work Item {workItem.PartitionIndex} (Execution time {totalExecutionTime})");
+                if (totalExecutionTime > s_maxExecutionTime)
+                {
+                    // Log a warning to the console with work item details when we were not able to partition in under our limit.
+                    // This can happen when a single specific test exceeds our execution time limit.
+                    ConsoleUtil.WriteLine($"##[warning]Work item {workItem.PartitionIndex} estimated execution {totalExecutionTime} time exceeds max execution time {s_maxExecutionTime}.");
+                    LogFilters(workItem, ConsoleUtil.WriteLine);
+                }
+                else
+                {
+                    LogFilters(workItem, Logger.Log);
+                }
             }
 
-            var typeRef = reader.GetTypeReference((TypeReferenceHandle)type.BaseType);
-            return
-                reader.GetString(typeRef.Namespace) == "System" &&
-                reader.GetString(typeRef.Name) is "Object" or "ValueType" or "Enum";
+            static void LogFilters(WorkItemInfo workItem, Action<string> logger)
+            {
+                foreach (var assembly in workItem.Filters)
+                {
+                    var assemblyRuntime = TimeSpan.FromMilliseconds(assembly.Value.Sum(f => f.ExecutionTime.TotalMilliseconds));
+                    logger($"    - {assembly.Key.AssemblyName} with execution time {assemblyRuntime}");
+                    var testFilters = assembly.Value;
+                    if (testFilters.Length > 0)
+                    {
+                        logger($"        - {testFilters.Length} tests: {string.Join(",", testFilters.Select(t => t.FullyQualifiedName))}");
+                    }
+                }
+            }
         }
 
-        private static string GetFullName(MetadataReader reader, TypeDefinition type)
+        private static ImmutableArray<TypeInfo> GetTypeInfoList(AssemblyInfo assemblyInfo)
         {
-            var typeName = reader.GetString(type.Name);
-
-            if (TypeAttributes.NestedPublic == (type.Attributes & TypeAttributes.NestedPublic))
+            var assemblyDirectory = Path.GetDirectoryName(assemblyInfo.AssemblyPath);
+            var testListPath = Path.Combine(assemblyDirectory!, "testlist.json");
+            Contract.Assert(File.Exists(testListPath));
```

Comment:
Please avoid `Contract` here. Either use `Debug.Assert` or do explicit throws.

Reply:
Will change, but for my education - why is `Contract` not desired?